File: HomeWork3/app/src/main/java/ru/tinkoff/ru/seminar/core/core/di/AppComponent.java
package ru.tinkoff.ru.seminar.core.core.di;

import javax.inject.Singleton;

import dagger.Component;
import ru.tinkoff.ru.seminar.core.core.di.module.AppModule;
import ru.tinkoff.ru.seminar.core.core.di.module.RetrofitModule;
import ru.tinkoff.ru.seminar.core.feature.presentation.MainViewModel;

@Singleton
@Component(modules = {AppModule.class, RetrofitModule.class})
public interface AppComponent {
    void inject(MainViewModel viewModel);
}

File: HomeWork4/app/src/main/java/com/memebattle/myapplication/core/di/AppComponent.kt
package com.memebattle.myapplication.core.di

import com.memebattle.myapplication.core.di.module.AppModule
import com.memebattle.myapplication.core.di.module.RoomModule
import com.memebattle.myapplication.feature.add.AddViewModule
import com.memebattle.myapplication.feature.list.ListViewModel
import com.memebattle.myapplication.feature.node.child.ChildViewModel
import com.memebattle.myapplication.feature.node.parents.ParentsViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, RoomModule::class])
interface AppComponent {
    fun inject(listViewModel: ListViewModel)
    fun inject(addViewModule: AddViewModule)
    fun inject(parentsViewModel: ParentsViewModel)
    fun inject(childViewModel: ChildViewModel)
}